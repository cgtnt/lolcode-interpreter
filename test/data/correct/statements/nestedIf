HAI 

I HAS A NAME ITZ "BOB"

VISIBLE SMOOSH "HELLO, " AN NAME MKAY!

BOTH SAEM NAME AN "ALICE", O RLY?
  YA RLY
    VISIBLE "WELCOME BACK, ALICE"!
  NO WAI
    BOTH SAEM NAME AN "BOB", O RLY?
      YA RLY
        VISIBLE "HEY BOB"!
      NO WAI
        VISIBLE "I DON'T KNOW U"!
    OIC
OIC

KTHXBYE
