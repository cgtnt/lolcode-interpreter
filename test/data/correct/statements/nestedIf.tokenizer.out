@1:T_BEGIN:HAI-@3:T_COMMAND_TERMINATOR:
-@3:T_DECLARE_VAR:I HAS A-@3:T_T_IDENTIFIER:NAME-@3:T_DECLARE_SET_VAR:ITZ-@3:T_T_STRING:"BOB"-@5:T_COMMAND_TERMINATOR:
-@5:T_WRITE_STDOUT:VISIBLE-@5:T_CONCAT:SMOOSH-@5:T_T_STRING:"HELLO, "-@5:T_AND:AN-@5:T_T_IDENTIFIER:NAME-@5:T_END_INF:MKAY-@5:T_BANG:!-@7:T_COMMAND_TERMINATOR:
-@7:T_EQUAL:BOTH SAEM-@7:T_T_IDENTIFIER:NAME-@7:T_AND:AN-@7:T_T_STRING:"ALICE"-@7:T_COMMAND_TERMINATOR:,-@7:T_IF:O RLY?-@8:T_COMMAND_TERMINATOR:
-@8:T_THEN:YA RLY-@9:T_COMMAND_TERMINATOR:
-@9:T_WRITE_STDOUT:VISIBLE-@9:T_T_STRING:"WELCOME BACK, ALICE"-@9:T_BANG:!-@10:T_COMMAND_TERMINATOR:
-@10:T_ELSE:NO WAI-@11:T_COMMAND_TERMINATOR:
-@11:T_EQUAL:BOTH SAEM-@11:T_T_IDENTIFIER:NAME-@11:T_AND:AN-@11:T_T_STRING:"BOB"-@11:T_COMMAND_TERMINATOR:,-@11:T_IF:O RLY?-@12:T_COMMAND_TERMINATOR:
-@12:T_THEN:YA RLY-@13:T_COMMAND_TERMINATOR:
-@13:T_WRITE_STDOUT:VISIBLE-@13:T_T_STRING:"HEY BOB"-@13:T_BANG:!-@14:T_COMMAND_TERMINATOR:
-@14:T_ELSE:NO WAI-@15:T_COMMAND_TERMINATOR:
-@15:T_WRITE_STDOUT:VISIBLE-@15:T_T_STRING:"I DON'T KNOW U"-@15:T_BANG:!-@16:T_COMMAND_TERMINATOR:
-@16:T_END_IF:OIC-@17:T_COMMAND_TERMINATOR:
-@17:T_END_IF:OIC-@19:T_COMMAND_TERMINATOR:
-@19:T_END:KTHXBYE-@20:T_COMMAND_TERMINATOR:
-@20:T_EOF:
